src/api/c/det.cpp

```diff
@@ -78,19 +78,30 @@ af_err af_det(double *real_val, double *imag_val, const af_array in)
 
         cfloat cfval;
         cdouble cdval;
+        bool empty = i_info.dims()[0] == 0;
 
         switch(type) {
-        case f32: *real_val = det<float  >(in);  break;
-        case f64: *real_val = det<double >(in);  break;
+        case f32: *real_val = (empty)? 1.f : det<float  >(in);  break;
```

Comment:
Perhaps do the checks in the wrapper function. That way you can return scalar<T>(1.0); once and not worry about multiple if/elese statements?